mapped_file.hpp

```diff
@@ -86,7 +87,10 @@ struct mapped_file
 template <typename T>
 struct mapped_vector
 {
-	enum { header_size = 16 };
+	// the header must be large enough to make the first element still be
+	// correctly aligned
+	static constexpr size_t header_size = std::max(size_t(16), alignof(T));
```

Comment:
std::max isn't constexpr in C++11, I don't think the qualifier is necessary in this case.